internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -958,6 +958,13 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 			},
 		},
 	}
+
+	schemaKubernetesAddOns(resource)
+
+	if !features.ThreePointOhBeta() {
+		resource.Schema["addon_profile"] = schemaKubernetesAddOnProfiles()
+	}
```

Comment:
(as above) we should combine these two and return a list of fields to add to the schema - this means the resource is only requesting these fields and isn't likely to be manipulated elsewhere (avoiding any potential bugs)